### PR TITLE
Merge lookup_fully_qualified and lookup_fully_qualified_or_none

### PR DESCRIPTION
**Pull request by 97littleleaf11**
### Description

Partially solves https://github.com/python/mypy/issues/4157

* `SemanticAnalyzerCoreInterface.lookup_fully_qualified` seems redundant

### CODE REVIEW

**Comment by 97littleleaf11** [comment]
@hauntsaninja Should we use `lookup_fully_qualified` instead of `lookup_fully_qualified_or_none`? There exists other `lookup_fully_qualified` functions that return optional types. Thus I get rid of `or_none` for consistency.

**Review by hauntsaninja — COMMENTED** [review]
The change and shorter name seem fine to me. It might be worth doing `lookup_fully_qualified_or_none = lookup_fully_qualified` in `SemanticAnalyzer`, just to reduce churn / mypy versioning headaches for plugins.

File: mypy/plugin.py [hunk 1]
```diff
@@ -303,20 +303,16 @@ def builtin_type(self, fully_qualified_name: str) -> Instance:
...
         """Lookup a symbol by its fully qualified name.
 
-        Raise an error if not found.
+        Return None if not found.
         """
         raise NotImplementedError
 
-    @abstractmethod
     def lookup_fully_qualified_or_none(self, name: str) -> Optional[SymbolTableNode]:
-        """Lookup a symbol by its fully qualified name.
```
Comment (97littleleaf11) [inline mypy/plugin.py:315]: @hauntsaninja I'm not sure whether it'a good way to keep an un-abstract method in an Interface.

File: mypy/semanal.py [hunk 2]
```diff
@@ -4362,23 +4362,6 @@ def create_getattr_var(self, getattr_defn: SymbolTableNode,
             return v
         return None
 
-    def lookup_fully_qualified(self, name: str) -> SymbolTableNode:
```
Comment (JukkaL) [inline mypy/semanal.py:4365]: In the above comment the idea would be to have the simple implementation that calls `lookup_fully_qualified_or_none` and does an assert here.

File: mypy/semanal_shared.py [hunk 3]
```diff
@@ -37,9 +37,10 @@ def lookup_qualified(self, name: str, ctx: Context,
                          suppress_errors: bool = False) -> Optional[SymbolTableNode]:
         raise NotImplementedError
 
-    @abstractmethod
     def lookup_fully_qualified(self, name: str) -> SymbolTableNode:
-        raise NotImplementedError
+        ret = self.lookup_fully_qualified_or_none(name)
+        assert ret is not None
+        return ret
```
Comment (JukkaL) [inline mypy/semanal_shared.py:43]: Similar to above, maybe we can keep this abstract if the simple implementation is in `mypy.semanal`.

**Review by JukkaL — APPROVED** [review]
Thanks for the updates! The PR looks nice and tidy now. Sorry for the extra iterations, but I think that the end result makes it worth it.

**Pull request merged**